web-console/src/lib/compositions/streaming/inspection/useDeleteRows.ts

```diff
@@ -0,0 +1,45 @@
+import useStatusNotification from '$lib/components/common/errors/useStatusNotification'
+import { ApiError, PipelinesService } from '$lib/services/manager'
+import { useCallback } from 'react'
+
+import { useMutation } from '@tanstack/react-query'
+
+type Args = [
+  pipelineId: string,
+  relation: string,
+  rows: Partial<Record<'insert' | 'delete', Record<string, unknown>>>[],
+  isArray?: boolean
+]
+
+export function useInsertDeleteRows() {
```

Comment:
why is it called InsertDeleteRows wouldn't `useDeleteRows` be a better name?

Reply:
I might need to come up with a better name yet - this specific function allows both insertion and deletion, and I don't want to artificially split it into two different hooks